private_sharing/migrations/0003_auto_20160909_0427.py

```diff
@@ -0,0 +1,99 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.9.9 on 2016-09-09 04:27
+from __future__ import unicode_literals
+
+from string import digits  # pylint: disable=deprecated-module
+
+from django.db import migrations
+
+from common.utils import generate_id
+
+
+def migrate_go_viral_files(apps, *args):
+    Member = apps.get_model('open_humans', 'Member')
+    DataFile = apps.get_model('data_import', 'DataFile')
+    OnSiteDataRequestProject = apps.get_model('private_sharing',
+                                              'OnSiteDataRequestProject')
+    DataRequestProjectMember = apps.get_model('private_sharing',
+                                              'DataRequestProjectMember')
+    ProjectDataFile = apps.get_model('private_sharing', 'ProjectDataFile')
+    UserData = apps.get_model('go_viral', 'UserData')
+
+    try:
+        rumi = Member.objects.get(user__username='rumichunara')
+    except Member.DoesNotExist:
+        return
+
+    def random_id():
+        code = generate_id(size=8, chars=digits)
+
+        while DataRequestProjectMember.objects.filter(
+                project_member_id=code).count() > 0:
+            code = generate_id(size=8, chars=digits)
+
+        return code
+
+    project = OnSiteDataRequestProject(
```

Comment:
I think an OAuth2 project would make more sense -- technically that's what it was (albeit custom). One advantage is that this doesn't include the "consent" form, which doesn't make sense in this context.